### ISSUE TITLE
Live Mail EML Import problem

### ISSUE THREAD

**Issue by Nancy948** [thundernest/import-export-tools-ng#29]
I am trying to import eml emails from my old computer with Windows LIve Mail to my new computer with Thunderbird Mail. (I am a fairly new TB user - about 6 months.) Ideally, I would like to import the entire folder structure. I've got the emails on a flash drive. I have downloaded and installed ImportExportToolsNG. I don't know how to make it work. I can't find any instructions.  

I assume that I need to have my cursor in the folder pane on the folder where I want the messages to be placed, because it never asks me where to put them.  I have created a folder in Local Folders and that is highlighted.

I have gone to Tools/IimportExportTools NG. I have tried "import messages" selecting just a few messages.  Nothing happens. I have tried "Import all messages from a directory."  Folders are moved but not the messages.  What am I doing wrong?

Windows 10, TB 68.2.0

Thanks for any help.

Nancy

**Comment by cleidigh** [comment]
@Nancy948 
Just saw your issue, thanks for posting and using Thunderbird!
I recognize that I probably need to bring up some instructions since there is a lot of functionality in this extension.  I have actually taken over the extension because the original author abandonment.
I am trying to keep up with the issues given the changes in Thunderbird my lack of familiarity.

You should be able to import using Import messages from directory and choose with subfolders.
I am assuming you are using the latest 4.0.4 version of the extension?
I had to fix several import-export issues before that.  
let me know if that works.  BTW probably not online for much longer...
cheers

**Comment by Nancy948** [comment]
I had been using 60. When I was writing the help! message initially, I went to check the release # and saw that TB was alerting me of an update, so I thought maybe that's my problem. It didn't work with 60 either!
Yes, I'm using 4.0.4.  
I'll be shutting down soon here myself.  
Thanks for your help.
nancy

**Comment by cleidigh** [comment]
okay good
I just did a quick test import successfully with 68.2 however this is using an export from the extension itself not live mail.  let me put together a couple test files which I can post in the repository.
I will do that tomorrow and post here with instructions to test.
thanks for your patience lots of moving parts

**Comment by cleidigh** [comment]
@Nancy948 
I managed to get LiveMail 2012 up and running under Windows  10.  I assume that's what you were using assess the last addition.  I exported the inbox of the test account I have been using for everything.  I chose the LiveMailFormat for the export.  I then successfully imported the inbox from the exported folder into Thunderbird 68.2 in a local folder.  I have done the same before with a single message as well.

Given the above and the fact that I think it's up to 25k users, my hope is it is some configuration issue.
I now have a test folder in the repository.  I added the exported folder for you to test, that way we are looking at the exact same thing.  

https://github.com/thundernest/import-export-tools-ng/tree/master/tests/import/LiveMail/test-inbox/Kokkini%20(test1)

**Comment by Nancy948** [comment]
I'm sorry but I need more instructions. I went to that link, but I don't know what to do on that page to get the files where I can import them. I tried clicking on the clipboard symbol.  It said "Copied" but when I went to my Windows Explorer, the Paste command was not available (grayed out).  I tried right-clicking on the three dots and also on the clipboard. I got "save as" but the only file options were html. 

I was using Windows Live Mail 2011 (Build 15.4.3555.0308). 

I'm pretty comfortable with computers -- been using them for 40 years and know my way around well enough to be the troubleshooter for many friends and family, but I am not a developer, don't code, and only got onto GitHub to be able to ask this question. I've never used test folders from a repository. So please excuse my lack of knowledge here.

Thanks for your help.

**Comment by cleidigh** [comment]
@Nancy948 
The truth is this is not the seamless way to do things and my apologies for being too brief in the description, github has lots of good things but it's not exactly set up to be a support site and there is no reason to expect most users to have familiarity with.  But I have to remind myself of that.

so my goal is to provide you with one or more test files.  we can start with just one 
- if you follow the link you will be in a subfolder with several .eml files.
-  those files are all text it can actually be viewed online
- its kind of indirect, but to download this in the file you click on the file link 
then it will show you its contents fits text,  if you right-click on the raw button you will get the browser pop-up to save as .
- save it to your file system and then we will try and import 
![image](https://user-images.githubusercontent.com/25272315/67916511-51449680-fb6c-11e9-80e2-0159ebbecd73.png)

Back within Thunderbird, select a folder to import to , right click for the extension menu and choose import  messages, choose the single eml file you downloaded
- If this still does not work we should check the debug log using 'Control-Shift-J'
that brings up the debug console and it might have warnings or errors we can learn something from

I think I need to use a compressed file that's easily downloadable as a single entity
for our purposes I think we can use a single file for now.
The other thing we can try in parallel For this if you email me some random (obviously not personal)
eml file from your original export so I can test that as well.
use the following address:
test1@kokkini.net

I am sure you are anxious to use your e-mail so we will do our best to get this working, I appreciate you working through it with me. 👍 
Christopher

**Comment by Nancy948** [comment]
Christopher,

Everything is working tonight!  I'm not sure what has changed, because I'm not aware of anything I am doing differently.  But something must be different.  Your test file imported just fine.  So I tried one of mine again.  And it came in fine.  And then I tried some folders and subfolders, and they came in fine. Well, mostly -- a few folders  got skipped, but I will go back and move them individually. I have exported-imported about half of my old messages, and am currently exporting the rest (some storage files). 

I still don't know what yesterday's problem was.  I tried it many times and no emails ever imported.  I even had a computer-savvy friend looking over my shoulder to see if I was making any obvious mistakes.  It's a mystery!

I truly appreciate everything you did.  Thank you for your time and your help, and for providing the add-on that makes it possible.

Nancy

**Comment by cleidigh** [comment]
@Nancy948 
That's excellent.  Problems that don't really represent problems or good, on the other hand not knowing what happened is so not great.  What may be the case is there was some issue working with large files.  Glad it's working now.
I will go ahead and close this.  Reviews are always appreciated!
cheers
Christopher

**Comment by cleidigh** [comment]
@Nancy948 
Any possibility you would consider doing a review on the Thunderbird site?
Thanks!

**Comment by cleidigh** [comment]
@Nancy948 thanks for the review!

**Comment by t4thfavor** [comment]
I am trying to import a couple of gigs of EML files from windows live 2011. The results of the import are one or two folders get created, the first folder will have tons of guid named EML files, but they all look like folders, and none of them open. If I pick an individual folder, it seems to come in fine.

**Comment by t4thfavor** [comment]
![image](https://user-images.githubusercontent.com/12970092/68548151-be033080-03b7-11ea-8881-e7345942806c.png)

**Comment by cleidigh** [comment]
@t4thfavor 
Sorry for the delayed response, I never got a notification probably because this is closed.
It is usually best to open new issue especially if you perceive it to be different.  I will open this as a new issue , I will start with the assumption it may be a large import issue
